src/main/java/io/nats/client/support/JwtUtils.java

```diff
@@ -0,0 +1,145 @@
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.MessageDigest;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Base64;
+
+import static io.nats.client.support.JsonUtils.beginJson;
+import static io.nats.client.support.JsonUtils.endJson;
+
+public class JwtUtils {
+    static class Nats implements JsonSerializable {
+        String issuerAccount;
+        String[] tags;
+
+        @Override
+        public String toJson() {
+            StringBuilder sb = beginJson();
+            JsonUtils.addField(sb, "issuer_account", issuerAccount);
+            JsonUtils.addStrings(sb, "tags", Arrays.asList(tags));
+            JsonUtils.addField(sb, "type", "user");
+            JsonUtils.addField(sb, "version", 2);
+            return endJson(sb).toString();
+        }
+    }
+
+    static class Claim implements JsonSerializable {
+        Duration exp;
+        long iat;
+        String iss;
+        String jti;
+        String name;
+        Nats nats;
+        String sub;
+
+        @Override
+        public String toJson() {
+            StringBuilder sb = beginJson();
+            if (exp != null && !exp.isZero() && !exp.isNegative()) {
+                long unixSeconds = exp.toMillis() / 1000;
+                JsonUtils.addField(sb, "exp", unixSeconds);
+            }
+            JsonUtils.addField(sb, "iat", iat);
+            JsonUtils.addFieldEvenEmpty(sb, "jti", jti);
+            JsonUtils.addField(sb, "iss", iss);
+            JsonUtils.addField(sb, "name", name);
+            JsonUtils.addField(sb, "nats", nats);
+            JsonUtils.addField(sb, "sub", sub);
+
+            return endJson(sb).toString();
+        }
+    }
+
+    static final String ENCODED_CLAIM_HEADER =
+            ToBase64Url("{\"typ\":\"JWT\", \"alg\":\"ed25519-nkey\"}");
+
+    static final String NATS_USER_JWT_FORMAT = "-----BEGIN NATS USER JWT-----\n" +
+            "%s\n" +
+            "------END NATS USER JWT------\n" +
+            "\n" +
+            "************************* IMPORTANT *************************\n" +
+            "    NKEY Seed printed below can be used to sign and prove identity.\n" +
+            "    NKEYs are sensitive and should be treated as secrets.\n" +
+            "\n" +
+            "-----BEGIN USER NKEY SEED-----\n" +
+            "%s\n" +
+            "------END USER NKEY SEED------\n" +
+            "\n" +
+            "*************************************************************";
+
+    public static String issueUserCreds(String accountSeed, String issuerAccount, String userSeed, String userKeyPub, String optionalName, Duration optionalExpiration, String... optionalTags) throws GeneralSecurityException, IOException {
+        String jwt = issueUserJWT(accountSeed, issuerAccount, userKeyPub, optionalName, optionalExpiration, optionalTags);
+        return String.format(NATS_USER_JWT_FORMAT, jwt, userSeed);
```

Comment:
Since this is including the `userSeed`, we should be returning `char[]` rather than `String`, since this is a security best practice. Note that we should avoid any intermediate `String`s in that process. A simple way to do this is to allocate a string of the same length as the userSeed, do the `String.format()` and then convert this to a `char[]`, and then overwrite the char array offsets with the actual user seed.

Reply:
My java is rusty, but I'm buying the wiping argument. (can't manually overwrite a string to erase the content)

As far as I see it, either move to []byte or get rid of the `issueUserCreds` calls, making sure to make `NATS_USER_JWT_FORMAT` public 
@brimworks what do you think about dripping `issueUserCreds` and providing the printf pattern instead?

perhaps call out the printf pattern in the doc to `issueUserJWT`

Reply:
Using char[] rather thing String is a security best practice. The problem with String is that the garbage collector works differently in regards to Strings, and strings may even be "intern"ed so they get reused. Anyways, if you all feel strongly about using String, that is fine, I'm just bringing this up as a Java best-practice.

Reply:
@brimworks What I mean is basically avoid the entire problem by not having `issueUserCreds` and in the documentation of `issueUserJWT` point at `NATS_USER_JWT_FORMAT` instead.

Reply:
Yeah. I do like that idea. @scottf do you have an opinion?

Reply:
I'm fine with whatever you guys want. If It satisfies Matthias's requirements then go for it. We can always add it in later, it's harder to take it back

Reply:
`issueUserCreds` was removed and the `NATS_USER_JWT_FORMAT` is public along with documentation on how to use it to create credentials.